bitcoin/src/hash_types.rs

```diff
@@ -94,4 +94,16 @@ mod newtypes {
 
     impl_hashencode!(FilterHash);
     impl_hashencode!(FilterHeader);
+
+    impl From<Txid> for TxMerkleNode {
+        fn from(txid: Txid) -> Self {
+            Self::from(txid.0)
+        }
+    }
+
+    impl From<Wtxid> for WitnessMerkleNode {
+        fn from(wtxid: Wtxid) -> Self {
+            Self::from(wtxid.0)
```

Comment:
So this is not strictly wrong but since I think we should remove `From<sha256d::Hash>` this would then need to become `Self(wtxid.0)` (same above). Feel free to change but can go in either way.

Reply:
@condution true, but the viewpoint we're moving toward is that `MerkleNode`s are a different kind of hash than `sha256d::Hash`es, even though they're computed with the same algorithm.